Index: mbc5.c
===================================================================
--- mbc5.c	(revisione 84)
+++ mbc5.c	(revisione 85)
@@ -18,7 +18,7 @@
 */
 
 
-inline uint8_t MBC5_MemoryRead(int memaddr) {
+uint8_t MBC5_MemoryRead(int memaddr) {
   if ((memaddr >= 0xA000) && (memaddr < 0xC000)) {        /* RAM bank n */
     return(MemoryBankedRAM[(CurRamBank << 13) + memaddr]);
   } else if ((memaddr >= 0x4000) && (memaddr < 0x8000)) {    /* ROM bank #n */
Index: mbc0.c
===================================================================
--- mbc0.c	(revisione 84)
+++ mbc0.c	(revisione 85)
@@ -1,11 +1,11 @@
 /*
-  --------------------------------------
-  This file is part of the zBoy project.
-  Copyright (C) Mateusz Viste 2010, 2011
-  --------------------------------------
- 
-  ROM-ONLY support
-*/
+ * --------------------------------------------------------------------------
+ * This file is part of the zBoy project.
+ * Copyright (C) Mateusz Viste 2010, 2011, 2012, 2013, 2014, 2015
+ * --------------------------------------------------------------------------
+ *
+ * ROM-ONLY support
+ */
 
 
 uint8_t MBC0_MemoryRead(int memaddr) {
@@ -24,7 +24,5 @@
 
 
 void MBC0_MemoryWrite(int memaddr, uint8_t DataByte) {
-  /*PRINT "MMU fatal error: Tried to write to an unknown memory address: "; HEX(memaddr)
-    QuitEmulator = 1 */
   MemoryMAP[memaddr] = DataByte;
 }
